### ISSUE TITLE
[Bug Report] body_com_state_w not written to scene after reset

### ISSUE THREAD

**Issue by Jackkert** [isaac-sim/IsaacLab#1756]
### Describe the bug

**EDIT:** 
The description below describes the problem but I now understand that this problem is solved by these timestep settings:
https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py#L364

However, commit 999c1e9ab93acb07c0e31acca7bfe796ebfd3ab4 reverted the deprecated API calls, causing the function below to call old reset_root_state_uniform, which uses the previously marked deprecated body_state_w API. I changed my body_state_w to body_com_state_w when the deprecation error was displayed with the new release, the latest commit now caused this inconsistency. I'm wondering why both options are available now and if one will be removed in the future?

Hi developers,
The Articulation class currently only has functions that write the root states to sim. So for example, upon reset, if configured, the reset_root_state_uniform function in omni/isaac/lab/envs/mdp/event.py currently calls these functions:
```asset.write_root_pose_to_sim(torch.cat([positions, orientations], dim=-1), env_ids=env_ids)```
```asset.write_root_velocity_to_sim(velocities, env_ids=env_ids)```

From my understanding, this writes the new root pose and velocity to the simulation and the ArticulationData buffer (in the case of an articulation) after the env has reset. However, this does not update the body_com_state_w buffer in ArticulationData. Meaning that after an environment has reset, the new root state is set, but the states of the other links is not. 

With the way how the ManagerBasedRLEnv (I'm referring to this function: https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_rl_env.py#L152) works, this means that the next time env.step() is called, the body_com_state_w is still the old buffer before the env was reset. The buffers are updated when scene.update(dt) is called, but that happens after the next time the policy is called in process_actions. This means that the action the policy outputs after an environment resets is based on the old body_com_state_w, for 1 environment step. In my case, my observations are retrieved from body_com_state_w, and so are the states later on in my control stack, so the agent will get a wrong reward signal based on the old body_com_state_w for 1 env step after reset. 

Is there a way to write body_com_state_w to the buffers right after the env has reset? Or update the scene of the reset envs separately after line 219?

### Steps to reproduce

- Get any ManagerBasedRLEnv
- In process_actions in the ActionTerm, log state for a link that is not the root link of your Articulation before the env is reset.
- Log the same state right after the env is reset for a link that is not the root link.
- The following states after that timestep should be updated after scene.update(dt) is called in https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_rl_env.py#L195

### System Info

Describe the characteristic of your environment:

- Commit: 157a19e29b0dcafd87abbdad6f2e9428df3f09be
- Isaac Sim Version: 4.2.0-rc.17+release.15988.c99988b1.gl%  
- OS: Ubuntu 22.04
- GPU: RTX 3090
- CUDA: 11.5
- GPU Driver: 535.183.01

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

- [x] body_com_state_w updates correctly after reset

**Comment by Jackkert** [comment]
The issue

**Comment by RandomOakForest** [comment]
Thank you for posting this. We will investigate.

**Comment by Jackkert** [comment]
Suggestion:
I believe adding 
```self._data._body_link_state_w.timestamp = -1.0
self._data._body_com_state_w.timestamp = -1.0```

Below https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab/isaaclab/assets/articulation/articulation.py#L337
May resolve the inconsistent use of body_state_w and body_com_state_w
